pkg/builder/types.go

```diff
@@ -80,6 +80,7 @@ type builderContext struct {
 	Build             v1.BuilderTask
 	BaseImage         string
 	Namespace         string
+	Integration       string
```

Comment:
I think this is breaking the original design that wants an IntegrationKit (and the related build) to be Integration agnostic in order to be reused. If you need the source at build time (which I suspect is the reason why you want to provide an Integration here), probably we need to think an alternative way of generating the project to be given to the builder and let also the runtime to manage that situation.

Reply:
If so it means that I need to have the sources in the Build model, do you confirm?

Reply:
You need to have the sources, but I am not sure where it does make sense to have them. The Build should probably know the entire project as a whole (including the sources, if required). That means that probably we need to split the generation of the project from the pure building in 2 separate concerns. Following this idea, once the project was generated (with source if it is a native build), the Build would takeover and proceed (natively or not). In any case, @astefanutti can give you some more detail as he had run some experiment when Joor dependency was not yet supported in native. Hopefully he can give some more concrete advice.